include/bout/petsclib.hxx

```diff
@@ -109,11 +119,13 @@ private:
 
 class PetscLib {
 public:
-  PetscLib() {}
+  PetscLib(Options* UNUSED(opt) = nullptr) {}
   ~PetscLib() {}
   
   static void setArgs(int &UNUSED(c), char** &UNUSED(v)) {}
   
+  void createKSPWithOptions(MPI_Comm& comm, KSP& ksp);
```

Comment:
I know this is the PETSc interface, but it should maybe be `KSP createKSPWithOptions(const MPI_Comm& comm)` instead.